packages/core/src/connectivity/scp-cf/destination/destination-service.ts

```diff
@@ -114,19 +116,40 @@ async function fetchDestinations(
     );
 }
 
+export interface AuthAndExchangeTokens {
+  authHeaderJwt: string;
+  exchangeHeaderJwt?: string;
+}
+
 /**
  * Fetches a specific destination by name from the given URI, including authorization tokens.
  * For destinations with authenticationType OAuth2SAMLBearerAssertion, this call will trigger the OAuth2SAMLBearerFlow against the target destination.
+ * In this pass the access token as string.
+ * Fetches a specific destination with authenticationType OAuth2UserTokenExchange by name from the given URI, including authorization tokens.
  *
  * @param destinationServiceUri - The URI of the destination service
- * @param jwt - The access token
+ * @param token - The access token or AuthAndExchangeTokens if you want to include the X-user-token for OAuth2UserTokenExchange.
  * @param destinationName - The name of the desired destination
  * @param options - Options to use by retrieving destinations
  * @returns A Promise resolving to the destination
  */
 export async function fetchDestination(
   destinationServiceUri: string,
-  jwt: string,
+  token: string | AuthAndExchangeTokens,
```

Comment:
1. Not entirely sure, but isn't that a breaking change?
2. Do we really need two functions here? Couldn't the token thingy be done on the first line?

Reply:
I think it is not breaking since the string input is still possible. I needed a way to allow to bring the `user exchange header` inside the call of the `callDestinationService`. However there is such a strong coupling to the cache that I did it this way. I think about something better for a while